src/lib/CMakeLists.txt

```diff
@@ -244,6 +244,18 @@ set_target_properties(librnp
 
 if (BUILD_SHARED_LIBS)
   add_library(librnp-static STATIC $<TARGET_OBJECTS:librnp-obj> $<TARGET_OBJECTS:rnp-common>)
+  # Limit symbols export only to rnp_* functions.
+  if (APPLE)
+    # Check for Apple OSs and use -export_symbols_list
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/librnp.symbols" "_rnp_*\n")
```

Comment:
I don't claim to know much about macOS, but i'm a bit surprised to see the leading underscore here.  maybe the leading underscore is injected by the macOS toolchain?  [Apple documentation](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/DynamicLibraryDesignGuidelines.html#//apple_ref/doc/uid/TP40002013-SW4) seems to suggest that it does, though.

Reply:
@dkg thanks for the link, I was searching from the other side (for replacement of version script). As my development machine is Mac, I used try-and-see approach for this underscore and proven that it works. Probably I should update tests to check for the exported symbols for the macOS CI as well...